courses-app/src/App.tsx

```diff
@@ -1,13 +1,144 @@
-import { StyledWrapper } from './App.style';
+import { useState, createContext } from 'react';
+import { Routes, Route, Navigate, useNavigate } from 'react-router-dom';
+
 import Courses from './components/Courses/Courses';
 import Header from './components/Header/Header';
+import NotFound from './components/NotFound';
+import CourseInfo from './components/CourseInfo';
+import CreateCourse from './components/CreateCourse';
+import Login from './components/Login/Login';
+import Registration from './components/Registration/Registration';
+
+import { mockedAuthorsList, mockedCoursesList } from './constants';
+
+import {
+	AuthorType,
+	CourseType,
+	GlobalSharedContext,
+	TokenResponse,
+} from './types';
+
+import { StyledWrapper } from './App.style';
+
+export const SharedContext = createContext<GlobalSharedContext | undefined>(
+	undefined
+);
 
 function App() {
+	const [token, setToken] = useState<string | null>(
+		localStorage.getItem('token')
+	);
+	const [userName, setUserName] = useState<string | null>(
+		localStorage.getItem('userName')
+	);
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [courses, setCourses] = useState<CourseType[]>(mockedCoursesList);
+	const [allAuthors, setAllAuthors] = useState<AuthorType[]>(mockedAuthorsList);
+
+	const navigate = useNavigate();
+
+	const handleLogin = ({ result, user }: TokenResponse) => {
+		localStorage.setItem('token', result);
+		localStorage.setItem('userName', user.name);
+		setToken(result);
+		setUserName(user.name);
+	};
+
+	const handleLogout = () => {
+		localStorage.removeItem('token');
+		localStorage.removeItem('userName');
+		setToken(null);
+		setUserName(null);
+	};
+
+	const creationCoursesHandler = (
+		course: CourseType,
+		courseAuthors: AuthorType[]
+	) => {
+		setCourses([...courses, course]);
+		setAllAuthors([
+			...allAuthors,
+			...courseAuthors.filter(
+				(atr) => !allAuthors.some((haveAuthor) => haveAuthor.id === atr.id)
+			),
+		]);
+		navigate('/courses');
+	};
+
 	return (
-		<StyledWrapper>
-			<Header />
-			<Courses />
-		</StyledWrapper>
+		<SharedContext.Provider
+			value={{
+				userName,
+				setUserName,
+				handleLogout,
+				token,
+				setToken,
+				isLoading,
+				setIsLoading,
+				courses,
+				setCourses,
+				allAuthors,
+				setAllAuthors,
+				creationCoursesHandler,
+				handleLogin,
+			}}
+		>
+			<StyledWrapper>
+				<Header />
+				<Routes>
+					<Route
+						path='/'
+						element={
+							token ? (
+								<Navigate to='/courses' />
+							) : (
+								<Navigate to='/login' replace />
+							)
+						}
+					/>
+					<Route
+						path='/courses'
+						element={token ? <Courses /> : <Navigate to='/login' replace />}
+					/>
+					<Route
+						path='/courses/add'
+						element={
+							token ? (
+								<CreateCourse
+								// onAdd={creationCoursesHandler}
+								// authors={allAuthors}
+								/>
+							) : (
+								<Navigate to='/login' replace />
+							)
+						}
```

Comment:
To not write `token ? :` every time, You can do it in `useEffect`. Then your `element` param will be much cleaner 

`useEffect(() => { !token && navigate(login) }, [token])`